eclipse-che/che-operator
# feat: Allow to set environment variables into Eclipse Che containers

Signed-off-by: Anatolii Bazko <abazko@redhat.com>



### What does this PR do?
Allow to set environment variables into Eclipse Che containers

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/21605

### How to test this PR?

1. Generate chectl templates:
 ```bash
 make gen-chectl-tmpl TARGET=/tmp/operator-resources 
```
2. Deploy  Eclipse Che
```
chectl server:deploy --platform=minikube --templates /tmp/operator-resources --che-operator-image=quay.io/abazko/operator:21605
``` 
3. Patch CheCluster CR
```
kubectl patch checluster/eclipse-che --patch '{"spec": {"components": {"dashboard": {"deployment": {"containers": [{"env": [{"name": "KUBERNETES_PORT_443_TCP", "value": "tcp://10.96.0.1:440"}]}]}}}}}' --type=merge -n eclipse-che
```
4. Wait some time until a new dashboard pod is run
5. Check env var
```
kubectl exec deploy/che-dashboard -n eclipse-che -- printenv KUBERNETES_PORT_443_TCP
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
